aten/src/ATen/core/TensorImpl.h

```diff
@@ -864,9 +856,35 @@ struct CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
   }
 
   inline void update_to_contiguous_strides() {
-    strides_.resize(0);
+    strides_.reset();
     is_contiguous_ = true;
   }
 
+public:
+  at::Storage storage_; // TODO: Fix visibility on me
+
+protected:
+  std::vector<int64_t> sizes_;
```

Comment:
Wouldn't it still be beneficial if we used `unique_ptr` for `sizes_` too? IIRC `vector` usually contains at least three 8B members: beginning pointer, end pointer, and allocated capacity. This means that currently we're using 32B for sizes and strides. If we were to keep an `int64_t dims` and two `unique_ptr`s, that would lower the size to 24B.

Reply:
Better still, if we allocate a single block for sizes and strides, then we only need to keep the pointer and the number of dimensions, pushing the size down to 16B 😄

Reply:
Yes, we can save another word this way, because we don't need the allocated data. However, I cannot actually apply this optimization right now, because Caffe2 Tensor API has a function `const std::vector<int64_t>& dims()`, which requires me to have an actual `std::vector` on hand. I am working on a refactor commit that removes the need from this, but in the meantime it's better to unblock this UBN.

And yes, allocating a single block is probably better.

Reply:
Hopefully it will be straightforward to switch C2 to `ArrayRef`s